corehq/apps/accounting/tests/test_domain_user_history.py

```diff
@@ -42,3 +49,86 @@ def test_calculate_users_in_all_domains(self):
         self.assertEqual(domain_user_history.num_users, self.num_users)
         self.assertEqual(domain_user_history.record_date, self.record_date)
+
+
+class TestBillingAccountWebUserHistory(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.billing_contact_enterprise = generator.create_arbitrary_web_user_name()
+        cls.billing_contact_standard = generator.create_arbitrary_web_user_name()
+
+        cls.dimagi_user = generator.create_arbitrary_web_user_name(is_dimagi=True)
+        cls.currency = generator.init_default_currency()
+        cls.account_enterprise = generator.billing_account(
+            cls.dimagi_user, cls.billing_contact_enterprise, is_customer_account=True)
+        cls.account_standard = generator.billing_account(
+            cls.dimagi_user, cls.billing_contact_standard, is_customer_account=True
+        )
+
+        cls.domain_1 = generator.arbitrary_domain()
+        cls.domain_2 = generator.arbitrary_domain()
+        cls.domain_3 = generator.arbitrary_domain()
+
+        subscription_start_date = datetime.date(2022, 3, 1)
+        enterprise_plan = generator.subscribable_plan_version(edition=SoftwarePlanEdition.ENTERPRISE)
+
+        cls.domain_1_enterprise_subscription = generator.generate_domain_subscription(
+            cls.account_enterprise,
+            cls.domain_1,
+            date_start=subscription_start_date,
+            date_end=None,
+            plan_version=enterprise_plan,
+            is_active=True,
+        )
+        cls.domain_2_enterprise_subscription = generator.generate_domain_subscription(
+            cls.account_enterprise,
+            cls.domain_2,
+            date_start=subscription_start_date,
+            date_end=None,
+            plan_version=enterprise_plan,
+            is_active=True,
+        )
+
+        cls.standard_subscription = generator.generate_domain_subscription(
+            cls.account_standard,
+            cls.domain_3,
+            date_start=subscription_start_date,
+            date_end=None,
+            is_active=True,
+        )
+
+        generator.arbitrary_webusers_for_domain(cls.domain_1.name, 2)
+        generator.arbitrary_webusers_for_domain(cls.domain_2.name, 2)
+        generator.arbitrary_webusers_for_domain(cls.domain_3.name, 2)
+
+    def test_calculate_web_users_for_enterprise_account(self):
+        tasks.calculate_web_users_in_all_billing_accounts()
+        enterprise_users = BillingAccountWebUserHistory.objects.get(
+            billing_account=self.account_enterprise
+        ).num_users
+
+        # Should have users from both domain_1 and domain_2
+        self.assertEqual(enterprise_users, 4)
+
+    def test_calculate_web_users_for_standard_account(self):
+        tasks.calculate_web_users_in_all_billing_accounts()
+        standard_users = BillingAccountWebUserHistory.objects.get(billing_account=self.account_standard).num_users
+
+        # Should only have users from both domain_2
```

Comment:
I guess this comment should be: `Should only have users from domain_3` ?

Reply:
Yes. you are correct @jingcheng16. Would you mind to get this comment fixed as part of your recent work?